### DISCUSSION TITLE
\footnotetext in latex table doesn't show up when rendered to HTML

### DISCUSSION THREAD

**Discussion by weberam2** [quarto-dev/quarto-cli#9949] [Q&A]
### Description
````markdown
::: {#tbl-mri}
```{=latex}
\begin{tabular*}{\textwidth}{@{}p{2.5cm}llp{2.5cm}p{2.5cm}@{}}
\toprule
    ~ & T1w & T2w & pcASL & SWI \\ \midrule
    Sequence & 3D FSPGR & 3D CUBE & Multi-shot 3D fast spin-echo & 3D spoiled GRE flow-compensated \\
    Phase-encoding direction & Coronal & Sagittal & Axial & Axial \\
    TR (ms) & 7.74 & 2,300 & 4,680 & 30.9 \\
    TE (ms) & 2.97 & 66.29 & 10.55 & 5.24 \\
    Flip angle & 12$^{\circ}$ & 90$^{\circ}$ & 111$^{\circ}$ & 20$^{\circ}$ \\
    FOV (cm) & 20 & 20 & 24 & 25 \\
    Acquisition matrix & 512 x 512 & 256 x 256 & 128 x 128 & 256 x 256 \\
    In-plane resolution (mm) & 0.39 x 0.39 & 0.78 x 0.78 & 1.875 x 1.875 & 0.977 x 0.977 \\
    Slice thickness (mm) & 1 & 1 & 4 & 2, reconstructed to 1 with zero filling (ZIP2) \\
    Number of slices & 126 & 106 & 50 & 92 \\
    Additional parameters & n/a & n/a & 1,450 ms label period; 2,025 ms pulse label; 24 control-label pairs & n/a \\
    Scan duration & 4 min 39 s & 5 min 1 s & 5 min 26 s & 5 min 29 s \\ \botrule
\end{tabular*}
\footnotetext{T1w = T1-weighted; T2w = T2-weighted; pcASL = pseudo-continuous arterial spin labelling; SWI = susceptibility weighted imaging; FSPG = fast spoiled gradient echo; CUBE = General Electric name of sequence, not an acronym; GRE = gradient echo; ZIP2 = through-plane zero filling interpolation}
```
Technical parameters for MR imaging pulse sequences
:::
````
I have the above in my index.qmd file

When I render to pdf, I see the footnote under my table just as I would like it.
However, if I render to HTML or MS Word it is gone.

Is there any way to get this to work for both?

PDF version:
![image](https://github.com/quarto-dev/quarto-cli/assets/16805029/290e36e5-188a-4cdf-8b7d-7b7089ee6c4c)

HTML version:
![image](https://github.com/quarto-dev/quarto-cli/assets/16805029/37966e5a-2366-4ed7-8925-8ea73b9ea10f)

I've tried formatting my tables in Markdown, HTML, R (kableExtras, gt) but I keep coming back to LaTeX as I feel I have the most control.

Any advice?

The other problem I was having was creating line-breaks in cells. I was using \makecell[tl] but that caused issues in HTML so I just gave up on controlling that...

**Comment by cderv** [comment]
> When I render to pdf, I see the footnote under my table just as I would like it.
> However, if I render to HTML or MS Word it is gone.

Your example (that I edited to [format correctly](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us)) only provide table as a Raw LaTeX block. So this block won't be shown in output other than LaTeX output (so `format: pdf`). 

Table and `\footnotetext` command are LaTeX and will be ignored in HTML or MS Word. Why do you expect any output ? Are you using any other conditional and table format ? 

> HTML version:

How did you get the HTML version ? 

> I've tried formatting my tables in Markdown, HTML, R (kableExtras, gt) but I keep coming back to LaTeX as I feel I have the most control.

You should have a look to **gt** which has table footnote element (https://gt.rstudio.com/reference/tab_footnote.html). **flextable** also have a function for this: https://davidgohel.github.io/flextable/reference/footnote.html

**Comment by weberam2** [comment]
Hi Christophe,

Thanks for fixing my formatting. I tried to use \```qmd but then the next \```, which should end \```{=latex}, ended up ending my \```qmd...

"Table and \footnotetext command are LaTeX and will be ignored in HTML or MS Word. Why do you expect any output ?"
I was not aware of this, and just expected the latex table to be able to convert to HTML or MS Word

"Are you using any other conditional and table format ?"
No, but this sounds like what I need...
Are there any resources to point to for me to learn about this option?

"How did you get the HTML version ?"

Here is my `_qaurto.yml` file:

```qmd
project:
  type: manuscript
  render: 
    - "*.qmd"
    - "!pyvenv_csvo/"

execute:
  freeze: auto

format:
  html:
    embed-resources: true
    toc: true
    comments:
      hypothesis: true
      
  docx: default
  jats: default
  
  nature-pdf:
    journal: "sn-nature"
    keep-tex: true

filters:
  - parse-latex
```

I get the HTML version when I preview (as it is default) and when I render.

I have looked at **gt**. While I understand (I think) how to make a table from an r dataframe, it's not as clear to me how to create a table like I would using simple markdown formatting or latex.

I know I can make a table in markdown, then load using `read.so` (https://rdrr.io/github/alistaire47/read.so/f/README.md)
But then I find my latex symbols don't work (for example, $^{\circ}$

Flextable looks excellent... but I think I may still have the issue with adding latex symbols...

**Reply by cderv** [reply]
> Thanks for fixing my formatting. I tried to use ```qmd but then the next ```, which should end ```{=latex}, ended up ending my ```qmd..

If you read our guide I linked to, you'll learn how to add more backticks to avoid this. 

> I was not aware of this, and just expected the latex table to be able to convert to HTML or MS Word

Yeah this is not a Pandoc feature or Quarto feature. What Quarto adds is HTML table parsing. Quarto processes HTML raw block with tables and converts to table in Pandoc to allow writing cross format https://quarto.org/docs/authoring/tables.html#html-tables

This does not happen with LaTeX. 

> "Are you using any other conditional and table format ?"
> No, but this sounds like what I need...
> Are there any resources to point to for me to learn about this option?

Look at conditional content in our website, and also what engine can do for you. Like **knitr** engine for R which allows to conditionnally compute based on format output. 

> filters:
>  - parse-latex

I see you are using a specific filter. I don't know it. So maybe this is what is parsing the table, but does not handle the footnotetext. You should open an issue to whomever provided the extension or the filter directly. 

Regarding table package options, you need to use the tool and language you are confortable with.
